src/code_80027040.c

```diff
@@ -1,189 +1,10 @@
 #include <ultra64.h>
 #include <macros.h>
+#include "main.h"
+#include "variables.h"
+#include "framebuffers.h"
 #include "common_structs.h"
-
-extern OSIoMesg gDmaIoMesg;
-extern OSMesgQueue gDmaMesgQueue;
-extern OSMesg gMainReceivedMesg;
-
-extern u8 _kart_texturesSegmentRomStart[];
-extern uintptr_t *gKartPalettes[];
-
-// arrays?
-extern u16 D_802F1F80[2][4][2048];
-extern u8 D_802DFB80[2][2][18688];
-
-extern u16 D_800DDEB0[];
-/*
-u16 D_800DDEB0[] = { // sizes
-    0x06c0, 0x06e0, 0x06e0, 0x0680, 0x07c0, 0x0700, 0x0680, 0x0910
-};
-*/
-extern s32 gActiveScreenMode;
-
-extern uintptr_t **gKartTextureGroup18s;
-
-extern uintptr_t *gKartMarioGroup8[];
```

Comment:
If this stuff should eventually be declared here (and not just extern'd) can this still be done in the .h file? or does it have to be in C?

Reply:
I don't understand the question.
If the question is:
In the future can the currently non-initialized be put in this .c file or do they have to go elsewhere?
Then I think the answer is they (and their extern'd declarations) have to have their own file

Reply:
Ah, ya, I see what's going on now. This change is good.

Eventually all of this data should be declared here. (Perhaps we could make a new file for it as long as it's linked just before its code file. This may or may not match like this). It's just sooo much data. I didn't do it all.